science/fetcher.py

```diff
@@ -23,6 +31,33 @@
 logger = logging.getLogger(__name__)
 
 
+retry_fetch = retry(
+    # Raise the final exception in a retry chain if all retries fail.
+    reraise=True,
+    retry=retry_if_exception(
+        lambda ex: (
+            isinstance(ex, TimeoutException)
```

Comment:
I tested most of this config by using a temporary low timeout on the httpx.Client. You do get 2 warning logs before the 2 retries and the final exception is the timeout of the last try.